MovieQuiz/Presentation/MovieQuizPresenter.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  MovieQuizPresenter.swift
+//  MovieQuiz
+//
+//  Created by Тадевос Курдоглян on 14.11.2022.
+//
+
+import Foundation
+import UIKit
+
+final class MovieQuizPresenter: QuestionFactoryDelegate {
+    private var currentQuestionIndex: Int = 0
+    var correctAnswers: Int = 0
+    private var questionsAmount: Int = 10
+    var currentQuestion: QuizQuestion?
+
+    var statisticService: StatisticServiceImplementation?
+    private weak var viewController: MovieQuizViewController?
+    private var questionFactory: QuestionFactoryProtocol?
+
+    init(viewController: MovieQuizViewControllerProtocol) {
+        self.viewController = viewController as? MovieQuizViewController
```

Comment:
Хороший паттерн, когда в инициализатор приходит не явная реализация, а ее интерфейс(протокол)

Принимая по протоколу, ты говоришь, что класс будет работать с некоторым объектом удовлетворяющим интерфейсу (протоколу), или иначе говорят объект конформит протоколу. Так мы сокрываем конечную реализацию, ограничиваем возможности доступа к управлению объектом, и получаем гибкость в использовании разных реализаций (хочешь настоящий контроллер, хочешь - только мок, хочешь другой контроллер). 

Каст и сохранение реального объекта – это ошибка этого принципа. 

Вот так будет верно:
```suggestion
    private weak var viewController: MovieQuizViewControllerProtocol?
    private var questionFactory: QuestionFactoryProtocol?

    init(viewController: MovieQuizViewControllerProtocol) {
        self.viewController = viewController
```

но это внесет те самые ограничения ( о них далее)